api/dialogflow_service.py

```diff
@@ -73,21 +77,29 @@ def add_entities(self, entity_name, entity_values):
 
     def get_entities(self):
         entity_types_client = dialogflow.EntityTypesClient()
-        return entity_types_client.list_entity_types(self.parent, DialogFlowService.LANGUAGE_CODE)
+        entities_iterator = entity_types_client.list_entity_types(self.parent, DialogFlowService.LANGUAGE_CODE)
+
+        entities = []
```

Comment:
I thought it would work with returning the iterator directly, why did you convert it to an array?

Also, a more pythonic way is with list comprehensions, it would look like this:
return [entity for entity in entities_iterator]

Just curious but you can merge if it does not work with returning the iterator

Reply:
ok, but would we keep the get_entities function name in that case? or would it become get_entities_iterator?